--- wasm_multithread/lib/cmake/Qt6/Qt6Targets.cmake
+++ wasm_multithread/lib/cmake/Qt6/Qt6Targets.cmake
@@ -63,7 +63,7 @@
   INTERFACE_COMPILE_OPTIONS "SHELL:-pthread"
   INTERFACE_INCLUDE_DIRECTORIES "${_IMPORT_PREFIX}/mkspecs/wasm-emscripten;${_IMPORT_PREFIX}/include"
   INTERFACE_LINK_LIBRARIES "embind;Threads::Threads"
-  INTERFACE_LINK_OPTIONS "SHELL:-s MAX_WEBGL_VERSION=2;SHELL:-s FETCH=1;SHELL:-s WASM_BIGINT=1;SHELL:-s MODULARIZE=1;SHELL:-s EXPORT_NAME=createQtAppInstance;SHELL:-s DISABLE_EXCEPTION_CATCHING=1;SHELL:-pthread;SHELL:-s ALLOW_MEMORY_GROWTH;\$<\$<CONFIG:Debug>:;SHELL:-s DEMANGLE_SUPPORT=1;--profiling-funcs>;SHELL:-sASYNCIFY_IMPORTS=qt_asyncify_suspend_js,qt_asyncify_resume_js;SHELL:-s ERROR_ON_UNDEFINED_SYMBOLS=1"
+  INTERFACE_LINK_OPTIONS "SHELL:-s MAX_WEBGL_VERSION=2;SHELL:-s FETCH=1;SHELL:-s WASM_BIGINT=1;SHELL:-s MODULARIZE=1;SHELL:-s EXPORT_NAME=createQtAppInstance;SHELL:-pthread;SHELL:-s ALLOW_MEMORY_GROWTH;\$<\$<CONFIG:Debug>:;SHELL:-s DEMANGLE_SUPPORT=1;--profiling-funcs>;SHELL:-sASYNCIFY_IMPORTS=qt_asyncify_suspend_js,qt_asyncify_resume_js;SHELL:-s ERROR_ON_UNDEFINED_SYMBOLS=1"
   _qt_package_version "6.5.2"
 )
 
--- wasm_multithread/lib/cmake/Qt6/QtWasmHelpers.cmake
+++ wasm_multithread/lib/cmake/Qt6/QtWasmHelpers.cmake
@@ -35,12 +35,8 @@
     endif()
 
     # wasm exceptions
-    if (QT_FEATURE_wasm_exceptions)
         target_compile_options("${wasmTarget}" INTERFACE -fwasm-exceptions)
         target_link_options("${wasmTarget}" INTERFACE -fwasm-exceptions)
-    else()
-        target_link_options("${wasmTarget}" INTERFACE "SHELL:-s DISABLE_EXCEPTION_CATCHING=1")
-    endif()
 
     if (QT_FEATURE_thread)
         target_compile_options("${wasmTarget}" INTERFACE "SHELL:-pthread")
